specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json

```diff
@@ -1241,629 +1286,2742 @@
+        "operationId": "Namespaces_ListNetworkRuleSets",
+        "description": "Gets list of NetworkRuleSet for a Namespace.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/namespaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of NetworkRuleSets for Namespace successfully returned.",
+            "schema": {
+              "$ref": "#/definitions/NetworkRuleSetListResult"
+            }
+          },
+          "default": {
+            "description": "EventHub error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
         }
       }
     },
-    "Operation": {
-      "description": "A Event Hub REST API operation",
-      "type": "object",
-      "properties": {
-        "name": {
-          "readOnly": true,
-          "type": "string",
-          "description": "Operation name: {provider}/{resource}/{operation}"
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventHub/namespaces/{namespaceName}/messagingplan": {
+      "get": {
+        "tags": [
+          "Namespaces"
+        ],
+        "operationId": "Namespaces_GetMessagingPlan",
+        "x-ms-examples": {
+          "GetNamespaceMessagingPlan": {
+            "$ref": "./examples/NameSpaces/EHNameSpaceGetMessagingPlan.json"
+          }
         },
-        "display": {
-          "description": "The object that represents the operation.",
-          "properties": {
-            "provider": {
-              "readOnly": true,
-              "type": "string",
-              "description": "Service provider: Microsoft.EventHub"
-            },
-            "resource": {
-              "readOnly": true,
-              "type": "string",
-              "description": "Resource on which the operation is performed: Invoice, etc."
-            },
-            "operation": {
-              "readOnly": true,
-              "type": "string",
-              "description": "Operation type: Read, write, delete, etc."
+        "description": "Gets messaging plan for specified namespace.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/namespaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
```

Comment:
Please include next link. If your service returns null for this field, that is perfectly fine. This will allow you to add paging later without a breaking change.

        "x-ms-pageable": {
          "nextLinkName": "nextLink"
        }

Reply:
we are deprecating this API, its supported in older versions. removed from swagger.